tests/func/test_add.py

```diff
@@ -496,27 +496,27 @@ def test_should_update_state_entry_for_directory_after_add(
 
     ret = main(["add", "data"])
     assert ret == 0
-    assert file_md5_counter.mock.call_count == 3
+    assert file_md5_counter.mock.call_count == 4
```

Comment:
This is because we previously called `stage` for every file in a directory when running `checkout()`, and now we just `stage` the whole directory in bulk, so that's +1 `file_md5` for `tree.digest()`.

Side note: we should revisit these kinds of tests and probably just have some unit tests instead.

Reply:
yeah, With `objects`, `dvc add` should not care about file_md5 at all.